StratisCore.UI/src/app/shared/services/address-book-service.ts

```diff
@@ -0,0 +1,57 @@
+import { Injectable } from '@angular/core';
+import { RestApi } from "@shared/services/rest-api";
+import { GlobalService } from "@shared/services/global.service";
+import { HttpClient, HttpParams } from "@angular/common/http";
+import { ErrorService } from "@shared/services/error-service";
+import { BehaviorSubject, Observable } from "rxjs";
+import { catchError } from "rxjs/operators";
+import { AddressLabel } from "@shared/models/address-label";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AddressBookService extends RestApi {
+  private contactsSubject = new BehaviorSubject<AddressLabel[]>([]);
+  public loading = new BehaviorSubject<boolean>(false);
+
+  constructor(globalService: GlobalService,
+              httpClient: HttpClient,
+              errorService: ErrorService) {
+    super(globalService, httpClient, errorService);
+    this.getContacts()
+  }
+
+  public get contacts(): Observable<AddressLabel[]> {
+    return this.contactsSubject.asObservable()
+  }
+
+  private getContacts(): void {
+    this.loading.next(true);
+    this.get<any>('addressBook').toPromise().then(result => {
+      this.contactsSubject.next(result.addresses);
+      this.loading.next(false)
+    });
+  }
+
+  public addAddressBookAddress(item: AddressLabel): Promise<any> {
```

Comment:
What makes you chose Promise over Observable?
I'm fine either way, but we should be consistent throughout the codebase.

Only adding this comment once - goes for all calls in services ofc.